packages/editor-ui/src/stores/workflows.ts

```diff
@@ -532,12 +532,16 @@ export const useWorkflowsStore = defineStore(STORES.WORKFLOWS, {
 			dataPinningEventBus.$emit('unpin-data', { [payload.node.name]: undefined });
 		},
 
-		addConnection(data: { connection: IConnection[] }): void {
+		addConnection(data: { connection: IConnection[]; setStateDirty: boolean }): void {
 			if (data.connection.length !== 2) {
 				// All connections need two entries
 				// TODO: Check if there is an error or whatever that is supposed to be returned
 				return;
 			}
+			const uiStore = useUIStore();
```

Comment:
@OlegIvaniv as a cleanup of the unwanted side effect of the node connection callback (see [my other comment](https://github.com/n8n-io/n8n/pull/5364/files#r1096826781) ) I made `uiStore.stateIsDirty` called everywhere before `workflowsStore.addConnection` is called so it wasn't necessary to set it here as well anymore